hikari/utilities/aio.py

```diff
@@ -89,3 +88,87 @@ def is_async_iterable(obj: typing.Any) -> bool:
     """Determine if the object is an async iterable or not."""
     attr = getattr(obj, "__aiter__", None)
     return inspect.isfunction(attr) or inspect.ismethod(attr)
+
+
+async def first_completed(
+    *aws: typing.Awaitable[typing.Any],
+    timeout: typing.Optional[float] = None,
+) -> None:
+    """Wait for the first awaitable to complete.
+
+    The awaitables that don't complete first will be cancelled.
+
+    Completion is defined as having a result or an exception set. Thus,
+    cancelling any of the awaitables will also result in the others being
+    cancelled.
+
+    If the first awaitable raises an exception, then that exception will be
+    propagated.
+
+    Parameters
+    ----------
+    *aws : typing.Awaitable[typing.Any]
+        Awaitables to wait for.
+    timeout : typing.Optional[float]
+        Optional timeout to wait for, or `builtins.None` to not use one.
+        If the timeout is reached, all awaitables are cancelled immediately.
+
+    !!! note
+        If more than one awaitable is completed before entering this call, then
+        the first future is always returned.
+    """
+    fs = list(map(asyncio.ensure_future, aws))
+    iterator = asyncio.as_completed(fs, timeout=timeout)
```

Comment:
this is probably anothing but would this look any cleaner using asyncio.wait?

Reply:
not really, asyncio.wait produces two lists you have to manually purge. I was trying to change this a few minutes ago but this was simpler as far as mypy was concerned, and I feel this represents what I am trying to do with it a little clearer.